apps/users/models.py

```diff
@@ -0,0 +1,101 @@
+    role_id = Column(INTEGER, ForeignKey("role.role_id"), nullable=True)
+
+    student = relationship("Student", back_populates="users")
+    user_request_reviews = relationship("UserRequestReview", back_populates="reviewer_user")
+    user_faculty = relationship("UserFaculty", back_populates="users")
+    user_request = relationship("UserRequest", back_populates="users")
+    roles = relationship("Role", back_populates="users")
+
+    def __repr__(self):
+        return f'{self.__class__.__name__}(user_id="{self.user_id}", login="{self.login}", password="{self.password}", ' \
+               f'last_visit="{self.last_visit}", email="{self.email}", is_active="{self.is_active}", role_id="{self.role_id}")'
+
+
+class OneTimeToken(Base):
+    __tablename__ = 'one_time_token'
+
+    token_id = Column(INTEGER, primary_key=True, nullable=False)
+    token = Column(VARCHAR(length=255), nullable=False)
+    expires = Column(TIMESTAMP, nullable=False)
+    student_id = Column(INTEGER, ForeignKey("student.student_id"), nullable=False)
+
+    student = relationship("Student", back_populates="one_time_token")
+
+    def __repr__(self):
+        return f'{self.__class__.__name__}(token_id="{self.token_id}",token="{self.token}",expires="{self.expires}",student_id="{self.student_id}")'
+
+
+class Student(Base):
+    __tablename__ = 'student'
+
+    student_id = Column(INTEGER, primary_key=True, nullable=False)
+    full_name = Column(VARCHAR(length=255), nullable=False)
+    telephone_number = Column(INTEGER, nullable=False, unique=True)
+    gender = Column(VARCHAR(length=1), nullable=False)
+    course_id = Column(INTEGER, ForeignKey("course.course_id"), nullable=False)
+    speciality_id = Column(INTEGER, ForeignKey("speciality.speciality_id"), nullable=False)
+    user_id = Column(INTEGER, ForeignKey("user.user_id"))
+    faculty_id = Column(INTEGER, ForeignKey("faculty.faculty_id"), nullable=False)
+
+    courses = relationship("Course", back_populates="student")
+    specialties = relationship("Speciality", back_populates="student")
+    users = relationship("User", back_populates="student")
+    faculties = relationship("Faculty", back_populates="student")
+    one_time_token = relationship("OneTimeToken", back_populates="student")
+
+    def __repr__(self):
+        return f'{self.__class__.__name__}(student_id="{self.student_id}",full_name="{self.full_name}",' \
+               f'telephone_number="{self.telephone_number}",gender="{self.gender}",' \
+               f'course_id="{self.course_id}",speciality_id="{self.speciality_id}",user_id="{self.user_id}",faculty_id="{self.faculty_id}")'
+
+
+class UserFaculty(Base):
+    __tablename__ = "user_faculty"
+
+    user_id = Column(INTEGER, ForeignKey("user.user_id"), nullable=False, primary_key=True)
+    faculty_id = Column(INTEGER, ForeignKey("faculty.faculty_id"), nullable=False, primary_key=True)
+
+    users = relationship("User", back_populates="user_faculty")
```

Comment:
This can be implemented via a secondary table.
Because of M2M relationship:
User -- UserFaculty (secondary) -- Faculty